src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_Call.cs

```diff
@@ -500,6 +597,144 @@ private static bool IsSafeForReordering(BoundExpression expression, RefKind kind
 
             Debug.Assert(actualArguments.All(static arg => arg is not null));
             return actualArguments.AsImmutableOrNull();
+
+            void visitInterpolatedStringArguments(ref BoundExpression? visitedReceiver, ImmutableArray<RefKind> argumentRefKindsOpt)
+            {
+                // There are handler conversions, so we need to do more complicated reordering of things. We need to
+                // put the receiver into a temp, and then store all the arguments up to the conversion into their own
```

Comment:
>and then store all the arguments [](http://example.com/codeflow?start=49&length=32)

Why all vs. just those used for handlers?

Reply:
Say we have the following call:

```cs
M(Arg1, Arg2, Arg3, Arg4, $"");
```

and that `Arg1` and `Arg3` are used in the string. We need to store `Arg1`, `Arg2`, and `Arg3`, in that order, to preserve lexical ordering. This original version would have also stored `Arg4` because detangling the stores in the single-pass version was too complex, but the new version I will push shortly will skip storing that 4th argument.